lib/request.js

```diff
@@ -433,12 +433,28 @@ module.exports = {
   get ips() {
     const proxy = this.app.proxy;
     const val = this.get(this.app.proxyIpHeader);
-    let ips = proxy && val
-      ? val.split(/\s*,\s*/)
-      : [];
+    let ips = [];
+
+    if (proxy && val) {
+      ips = val.split(/\s*,\s*/)
+        .map(host => {
+          let normalizedHost = host;
+
+          if (net.isIPv6(host)) {
+            normalizedHost = `[${host}]`;
```

Comment:
Because since this already validates it is the IP only, I would suggest simply shortcutting the rest of the processing for this case:
```suggestion
            return host;
```